test/v2/client/test_client_setup.go

```diff
@@ -0,0 +1,240 @@
+		return fmt.Errorf("failed to create test data directory: %w", err)
+	}
+
+	// Create the SRS directories if they do not exist
+	srsPath, err := config.Path(SRSPath)
+	if err != nil {
+		return fmt.Errorf("failed to get SRS path: %w", err)
+	}
+	err = os.MkdirAll(srsPath, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create SRS directory: %w", err)
+	}
+	srsTablesPath, err := config.Path(SRSPathSRSTables)
+	if err != nil {
+		return fmt.Errorf("failed to get SRS tables path: %w", err)
+	}
+	err = os.MkdirAll(srsTablesPath, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create SRS tables directory: %w", err)
+	}
+
+	// If any of the srs files do not exist, download them.
+	filePath, err := config.Path(SRSPathG1)
+	if err != nil {
+		return fmt.Errorf("failed to get SRS G1 path: %w", err)
+	}
+	_, err = os.Stat(filePath)
+	if os.IsNotExist(err) {
+		command := make([]string, 3)
+		command[0] = "wget"
+		command[1] = "https://srs-mainnet.s3.amazonaws.com/kzg/g1.point"
+		command[2] = "--output-document=" + filePath
+		logger.Info("executing %s", command)
+
+		cmd := exec.Command(command[0], command[1:]...)
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err = cmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to download G1 point: %w", err)
+		}
+	} else {
+		if err != nil {
+			return fmt.Errorf("failed to check if G1 point exists: %w", err)
+		}
+	}
+
+	filePath, err = config.Path(SRSPathG2)
+	if err != nil {
+		return fmt.Errorf("failed to get SRS G2 path: %w", err)
+	}
+	_, err = os.Stat(filePath)
+	if os.IsNotExist(err) {
+		command := make([]string, 3)
+		command[0] = "wget"
+		command[1] = "https://srs-mainnet.s3.amazonaws.com/kzg/g2.point"
+		command[2] = "--output-document=" + filePath
+		logger.Info("executing %s", command)
+
+		cmd := exec.Command(command[0], command[1:]...)
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err = cmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to download G2 point: %w", err)
+		}
+	} else {
+		if err != nil {
+			return fmt.Errorf("failed to check if G2 point exists: %w", err)
+		}
+	}
+
+	filePath, err = config.Path(SRSPathG2PowerOf2)
+	if err != nil {
+		return fmt.Errorf("failed to get SRS G2 power of 2 path: %w", err)
+	}
+	_, err = os.Stat(filePath)
+	if os.IsNotExist(err) {
+		command := make([]string, 3)
+		command[0] = "wget"
+		command[1] = "https://srs-mainnet.s3.amazonaws.com/kzg/g2.point.powerOf2"
+		command[2] = "--output-document=" + filePath
+		logger.Info("executing %s", command)
+
+		cmd := exec.Command(command[0], command[1:]...)
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		err = cmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to download G2 power of 2 point: %w", err)
+		}
+	} else {
+		if err != nil {
+			return fmt.Errorf("failed to check if G2 power of 2 point exists: %w", err)
+		}
+	}
+
+	// Check to see if the private key file exists. If not, stop the test.
+	filePath, err = ResolveTildeInPath(config.KeyPath)
```

Comment:
nit: consider moving this check in front of the SRS downloads, since those take a long time, but this is quick

Reply:
done